packages/provider_devtools_extension/lib/src/instance_viewer/instance_providers.dart

```diff
@@ -406,7 +406,7 @@ Future<List<ObjectField>> _parseFields(
     final owner =
         await eval.safeGetClass(fieldDeclaration.owner! as ClassRef, isAlive);
 
-    final ownerUri = fieldDeclaration.location!.script!.uri!;
+    final ownerUri = owner.location?.script?.uri ?? '';
```

Comment:
Why this change?

Reply:
Yeah, great question, should've take notes for myself :( But, story is like this: `fieldDeclaration` (`field.decl`) is deprecated field, and I thought it's great idea to start getting rid of it's usages. Apart of this, while I was testing, it was always null. Plus, it's kinda strange to gather location of field itself here, and not owner – I understand that they should not differ in this case, but from semantics point view it seems odd for me.

But! I will reconsider this part anyway, will come back with proper explanation at least.

Reply:
First, IMO switching from `!.` to `?.   ?? ''` isn't right. `''` isn't a valid URI and it's just going to silently fail later.
We should fail early.


We do care about where the field is located, not where the owner is located. An owner and a field may be in a different location (as a field may be inherited by a superclass or a mixin).

The logic here is trying to write code to read a field. As such, it is mandatory to run the read operation in the context of where the field is defined. This is for one core reason: The field may be private.

---

Ultimately, the variable name is likely wrong here. It should be `fieldUri` not `ownerUri`.